sdk/src/test/java/com/google/cloud/dataflow/sdk/io/CompressedSourceTest.java

```diff
@@ -466,6 +466,7 @@ public boolean producesSortedKeys(PipelineOptions options) throws Exception {
 
       public ByteReader(ByteSource source) {
         super(source);
+        offset = source.getStartOffset() - 1;
```

Comment:
We were always defaulting to -1, which is wrong if the range does not start at 0.

As for why -1 -- the easiest implementation here is to increment offset in every call to read a record, so they all start 1 before the first offset so that after the first record, getCurrentOffset is accurate.